commit cc32667c5d7d9ff95e65cc21f59905d8f9218394
Author: Diego Escalante Urrelo <diegoe@gnome.org>
Date:   Thu Oct 31 05:51:22 2019 -0500

    build: Update use of link_whole for meson-0.52
    
    A regression in meson-0.52 caused uses of link_whole to expose scenarios
    where duplicate symbols issues could appear. In particular
    libdconf_client_dep was being link_whole'd to itself, which recursively
    already included libdconf_common which was also a link_whole.
    
    This change does not modify the available symbols in libdconf.so, and is
    compatible with meson-0.52 and 0.51.
    
    See: https://github.com/mesonbuild/meson/pull/6030
    Fixes: https://gitlab.gnome.org/GNOME/dconf/issues/59

diff --git a/client/meson.build b/client/meson.build
index f3b7122..de6387e 100644
--- a/client/meson.build
+++ b/client/meson.build
@@ -28,7 +28,7 @@ libdconf_client = static_library(
 
 libdconf_client_dep = declare_dependency(
   dependencies: gio_dep,
-  link_whole: libdconf_client,
+  link_with: libdconf_client,
 )
 
 libdconf = shared_library(
